docs/module.rst

```diff
@@ -0,0 +1,267 @@
+.. _module-page:
+
+*******
+Modules
+*******
+
+.. note::
+    Modules were introduced in DSL2. If you are still using DSL1, see the :ref:`dsl1-page` page to
+    learn how to migrate your Nextflow pipelines to DSL2.
+
+In Nextflow, a **module** is a script that may contain functions, processes, and workflows. A module
+can be included in other modules or pipeline scripts and even shared across workflows.
+
+.. note::
+    Functions, processes, and workflows are collectively referred to as *components*.
+
+
+Module inclusion
+----------------
+
+A component defined in a module script can be imported into another Nextflow script using the ``include`` keyword.
+
+For example::
+
+    include { foo } from './some/module'
+
+    workflow {
+        data = channel.fromPath('/some/data/*.txt')
+        foo(data)
+    }
+
+The above snippet imports a process named ``foo``, defined in the module script, into the main
+execution context. This way, ``foo`` can be invoked in the ``workflow`` scope.
+
+Nextflow implicitly looks for the script file ``./some/module.nf``, resolving the path
+against the *including* script location.
+
+Module includes are subject to the following rules:
+
+- Relative paths must begin with the ``./`` prefix.
+- Include statements are not allowed from within a worklfow. They must occur at the script level.
+
+.. _module-directory:
+
+Module directory
+----------------
+
+.. note::
+    This feature requires Nextflow version ``22.10.0`` or later.
+
+A module can be defined as a directory with the same name as the module and with a script
+named ``main.nf``. For example::
+
+    some
+        └-module
+            └-main.nf
+
+When defined as a directory, the module must be included by specifying the module directory path::
+
+    include { foo } from './some/module'
+
+Module directories allow the use of module scoped binaries scripts. See `Module binaries`_ for details.
+
+Multiple inclusions
+-------------------
+
+A Nextflow script can include any number of modules, and an ``include`` statement can import any number of
+components from a module. Multiple components can be included from the same module by using the syntax
+shown below::
+
+    include { foo; bar } from './some/module'
+
+    workflow {
+        data = channel.fromPath('/some/data/*.txt')
+        foo(data)
+        bar(data)
+    }
+
+
+.. _module-aliases:
+
+Module aliases
+--------------
+
+When including a module component, it's possible to specify an *alias* with the ``as`` keyword.
+Aliasing allows you to include and invoke components with the same name, by assigning them different
+names in the including context. For example::
+
+    include { foo } from './some/module'
+    include { foo as bar } from './other/module'
+
+    workflow {
+        foo(some_data)
+        bar(other_data)
+    }
+
+You can even include the same component multiple times under different names::
+
+    include { foo; foo as bar } from './some/module'
```

Comment:
Ooh nice, I've never seen this done in one line before! I have just duplicated the include line a bunch of times.. 🤩

Reply:
*furiously updates all of the include statements across nf-core*